contracts/FindMarketAuctionEscrow.cdc

```diff
@@ -30,9 +30,9 @@ pub contract FindMarketAuctionEscrow {
 		access(contract) var auctionStartedAt: UFix64?
 		access(contract) var auctionEndsAt: UFix64?
 		access(contract) var offerCallback: Capability<&MarketBidCollection{MarketBidCollectionPublic}>?
+		access(contract) let saleItemExtraField: {String : AnyStruct}
 
-
-		init(pointer: FindViews.AuthNFTPointer, vaultType: Type, auctionStartPrice:UFix64, auctionReservePrice:UFix64, auctionDuration: UFix64, extentionOnLateBid:UFix64, minimumBidIncrement:UFix64) {
+		init(pointer: FindViews.AuthNFTPointer, vaultType: Type, auctionStartPrice:UFix64, auctionReservePrice:UFix64, auctionDuration: UFix64, extentionOnLateBid:UFix64, minimumBidIncrement:UFix64, saleItemExtraField: {String : AnyStruct}) {
```

Comment:
I think this also needs a validUntil. That is a timestamp on how long an auction listing is valid until you cannot start it. The reservePrice fields might get outdated so this will add more security.